source/internal.d.ts

```diff
@@ -76,6 +77,16 @@ Matches non-recursive types.
 */
 export type NonRecursiveType = BuiltIns | Function | (new (...args: any[]) => unknown);
 
+/**
+Returns a boolean for whether the given type is a normal key-value object.
+*/
+export type IsRecordObject<T> =
+	T extends NonRecursiveType | UnknownArray | ReadonlyMap<unknown, unknown> | ReadonlySet<unknown>
```

Comment:
What about WeakMap and other types? I feel like it would be better to come up with a check that didn't need an "exclude list" of types.